File: ios/Sources/NetworkInfoPlugin/NetworkInfoPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(NetworkInfoPlugin)
public class NetworkInfoPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "NetworkInfoPlugin"
    public let jsName = "NetworkInfo"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = NetworkInfo()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}

File: android/src/main/java/com/yangyang/networkinfo/NetworkInfoPlugin.java
package com.yangyang.networkinfo;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "NetworkInfo")
public class NetworkInfoPlugin extends Plugin {

    private NetworkInfo implementation = new NetworkInfo();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod
    public void getCurrentIp(PluginCall call) {
        String ip = implementation.getCurrentIp();
        JSObject ret = new JSObject();
        ret.put("ipAddress", ip);
        call.resolve(ret);
    }
}
